src/main/java/carpet/patches/EntityPlayerMPFake.java

```diff
@@ -187,6 +187,7 @@ public void die(DamageSource cause)
         shakeOff();
         super.die(cause);
         setHealth(20);
+        giveExperienceLevels(-(experienceLevel + 1));
```

Comment:
This line, which removes all levels and XP, should be wrapped in a check that `/gamerule keepInventory` is false.